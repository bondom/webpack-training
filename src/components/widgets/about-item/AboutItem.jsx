import React, {Component} from 'react';
import './AboutItem.scss';

class AboutItem extends Component {
    constructor(props){
        super(props);
        this.state = {className: "normal"};
        this.onMouseEnter = this.onMouseEnter.bind(this);
        this.onMouseLeave = this.onMouseLeave.bind(this);
    }

    onMouseEnter() {
        this.setState({className: "unnormal"});
    }

    onMouseLeave() {
        this.setState({className: "normal"});
    }

    render() {
        return (
            <div
                className={this.state.className}
                onMouseEnter={this.onMouseEnter}
                onMouseLeave={this.onMouseLeave}>
                    About item About item About item
            </div>
        )
    }
}

AboutItem.propTypes = {};

export default AboutItem;