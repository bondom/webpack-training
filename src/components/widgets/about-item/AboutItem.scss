.unnormal {
  background-color: white;
}

.hovered {
  background-color: antiquewhite;
}