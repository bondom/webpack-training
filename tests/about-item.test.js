import React from 'react';
import renderer from 'react-test-renderer';
import AboutItem from '../src/components/widgets/about-item/AboutItem';

test('Div changes class when hovered', () => {
    const component = renderer.create(
        <AboutItem/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    //manually trigger callback
    tree.props.onMouseEnter();
    //re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    //manually trigger callback
    tree.props.onMouseLeave();
    //re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})