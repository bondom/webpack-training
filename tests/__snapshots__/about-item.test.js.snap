// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Div changes class when hovered 1`] = `
<div
  className="normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  About item About item About item
</div>
`;

exports[`Div changes class when hovered 2`] = `
<div
  className="unnormal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  About item About item About item
</div>
`;

exports[`Div changes class when hovered 3`] = `
<div
  className="normal"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  About item About item About item
</div>
`;
